### DISCUSSION TITLE
Error when loading certificates

### DISCUSSION THREAD

**Discussion by ruimelodev** [MatthiasValvekens/pyHanko#345] [Q&A]
Hi @MatthiasValvekens,

I'm trying to load some certificates into my Signer but I'm always getting an error of a missing attribute:
```python
  File ".../python3.10/site-packages/pyhanko_certvalidator/registry.py", line 138, in register
    if cert.issuer_serial in self.certs:
AttributeError: 'generator' object has no attribute 'issuer_serial'
```

If I'm seeing it correctly, it seems the certificates have this info in `cert.issuer.serial_number` and not in `cert.issuer_serial`.

This is the code I'm using to load the certificates through `cert_registry` param:
```python
ExternalSigner(
    ama_client=ama_client,
    signing_cert=load_certs_from_pemder_data(b64decode(certificates[0])),
    signature_mechanism=algos.SignedDigestAlgorithm(
        {'algorithm': 'sha256_rsa'}
    ),
    cert_registry=SimpleCertificateStore.from_certs(
        [load_certs_from_pemder_data(b64decode(certificates[1])),
         load_certs_from_pemder_data(b64decode(certificates[2]))]
    ),
)
```

I'm not sure if it makes a difference but one of the certificates is a root certificate.

**Comment by MatthiasValvekens** [comment]
The problem is that that load_... call returns a generator, so you're passing a list of generators to a method that expects a list of certs. You'll want to chain those generators together instead of putting them in a list.
